gitlab/resource_gitlab_project_approval_rule.go

```diff
@@ -0,0 +1,425 @@
+package gitlab
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	gitlab "github.com/xanzy/go-gitlab"
+)
+
+// https://docs.gitlab.com/ee/api/merge_request_approvals.html#create-project-level-rule
+
+var resourceGitLabProjectApprovalRuleName = "gitlab_project_approval_rule"
+
+var resourceGitLabProjectApprovalRuleSchema = map[string]*schema.Schema{
+	"project": {
+		Type:     schema.TypeString,
+		ForceNew: true,
+		Required: true,
+	},
+	"name": {
+		Type:     schema.TypeString,
+		ForceNew: true,
+		Required: true,
+	},
+	"approvals_required": {
+		Type:     schema.TypeInt,
+		Required: true,
+	},
+	"user_ids": {
+		Type:     schema.TypeSet,
+		Optional: true,
+		Elem:     &schema.Schema{Type: schema.TypeInt},
+		Set:      schema.HashInt,
+	},
+	"group_ids": {
+		Type:     schema.TypeSet,
+		Optional: true,
+		Elem:     &schema.Schema{Type: schema.TypeInt},
+		Set:      schema.HashInt,
```

Comment:
should we set ContflictsWith ? or you can set both?

Reply:
You can set both or none. https://docs.gitlab.com/ee/api/merge_request_approvals.html#create-project-level-rule